Assets/Scripts/Attributes/SingleLayer.cs

```diff
@@ -0,0 +1,6 @@
+﻿using UnityEngine;
+
+public class SingleLayer : PropertyAttribute
```

Comment:
Attribute types should end with the suffix `Attribute`

Suggest renaming this type to `SingleLayerAttribute` and editor to `SingleLayerAttributeDrawer`